src/helpers.rs

```diff
@@ -54,6 +61,43 @@ pub fn copy_dir(source: &Path, dest: &Path) -> Result<()> {
     Ok(())
 }
 
+/// Find the root zine file in current dir
+fn zine_file_existed(path: &PathBuf) -> bool {
+    for entry in WalkDir::new(path).max_depth(1) {
+        let entry = entry.unwrap();
+        if entry.path().is_file() {
+            let root_file = entry.file_name().to_str().unwrap();
+            // Insure the name
+            if !root_file.eq(crate::ZINE_FILE) {
+                continue;
+            }
+            // Verify the content
+            let content = std::fs::read_to_string(path.join(root_file)).unwrap();
+            if toml::from_str::<Zine>(&content).is_err() {
```

Comment:
I suggest returning the deserialized `Zine` instance to avoid the second parse in the later building stage

Reply:
However, if we do so, we should rename the function `zine_file_existed` to another.

Reply:
I'm a little sory that at present I can't find a graceful way to  return the deserialized `Zine` instance which can avoid the second parse, because I find that with looping in `build`, the `Zine` instance seemed to update also...

Reply:
It's ok.